### ISSUE TITLE
multiple blocks with same pipelineId results in error

### ISSUE THREAD

**Issue by zybreak** [zont/gulp-usemin#125]
Having multiple blocks with the same pipelineId crashes gulp-usemin 0.3.11 with the following stacktrace:
```
stream.js:94
      throw er; // Unhandled stream error in pipe.
            ^
Error: write after end
    at writeAfterEnd (node_modules/gulp-minify-css/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:144:12)
    at DestroyableTransform.Writable.write (node_modules/gulp-minify-css/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:192:5)
    at Stream.ondata (stream.js:51:26)
    at Stream.emit (events.js:107:17)
    at Stream.endStream (node_modules/gulp-usemin/node_modules/gulp-concat/index.js:90:10)
    at _end (node_modules/gulp-usemin/node_modules/gulp-concat/node_modules/through/index.js:65:9)
    at Stream.stream.end (node_modules/gulp-usemin/node_modules/gulp-concat/node_modules/through/index.js:74:5)
    at Transform.onend (node_modules/gulp-usemin/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:523:10)                       
    at Transform.g (events.js:199:16)
    at Transform.emit (events.js:129:20)
```

Multiple blocks with same pipelineId seems to work fine in 0.3.8

**Comment by erikkemperman** [comment]
Same here; although in my case 0.3.11 did not throw an error but it generated duplicates of the same file (with different filenames due to rev) -- once for each block with the same pipelineId, I suspect. The other block with that pipelineId was not processed at all.

0.3.8 does what I expect.

**Comment by Snugug** [comment]
This is a problem for me as well. Reverting to 0.3.8 removes this error (but also prevents items with multiple of the same block from passing through).

Progress on this?

**Comment by zont** [comment]
Fixed in 0.3.14

**Comment by lcrodriguez** [comment]
I had to rollback to that version 0.3.8 because it did not work for me on the 0.3.14
I have two blocks named "js" as the pipelineId in the html file.

```javascript
gulp.task('dist:usemin', function() {
  return gulp.src([path.join(config.web.path.dist,'index.html')])
    .on('error', function(error) {
      plugins.log(error);
    })
    .pipe(plugins.usemin({
      css: [ plugins.stripCssComments({ preserve: false }), plugins.minifyCss({ processImport: false }), plugins.rev() ],
      html: [ function () { return plugins.minifyHtml({ empty: true }); } ],
      js: [ plugins.uglify(), plugins.rev() ]
    }))
    .pipe(gulp.dest(config.web.path.dist));
});
```
